src/bp/core/services/converse.ts

```diff
@@ -0,0 +1,132 @@
+import { UserRepository } from 'core/repositories'
+import { TYPES } from 'core/types'
+import { InvalidParameterError } from 'errors'
+import { EventEmitter2 } from 'eventemitter2'
+import { inject, injectable, postConstruct } from 'inversify'
+import _ from 'lodash'
+
+import { Event } from '../sdk/impl'
+
+import { EventEngine } from './middleware/event-engine'
+
+export const converseApiEvents = new EventEmitter2({ maxListeners: 100 })
+
+@injectable()
+export class ConverseService {
+  jsonMap: Map<string, object> = new Map()
+
+  constructor(
+    @inject(TYPES.EventEngine) private eventEngine: EventEngine,
+    @inject(TYPES.UserRepository) private userRepository: UserRepository
+  ) {}
+
+  @postConstruct()
+  init() {
+    this.eventEngine.register({
+      name: 'converse.capture.payload',
+      description: 'Captures the response payload for the Converse API',
+      order: 10000,
+      direction: 'outgoing',
+      handler: (event, next) => {
```

Comment:
event.channel === 'api'  to skip processing the api middleware for other channels